qctrlopencontrols/driven_controls/driven_control.py

```diff
@@ -594,6 +602,29 @@ def __str__(self):
 
         return driven_control_string
 
+    def __repr__(self):
+
+        """Returns a string representation for the object. The returned string looks like a valid
+        Python expression that could be used to recreate the object, including default arguments.
+
+        Returns
+        -------
+        str
+            String representation of the object including the values of the arguments.
+        """
+
+        attributes = {
+            'rabi_rates': self.rabi_rates,
+            'azimuthal_angles': self.azimuthal_angles,
+            'detunings': self.detunings,
+            'durations': self.durations,
+            'name': self.name
+        }
+
+        class_name = '{0.__class__.__name__!s}'.format(self)
```

Comment:
Rather than requiring each class do this manually, could we change create_repr_from_attributes to take the object itself and do this formatting internally?

Reply:
I like that too; in that case we can also supply a list of attributes instead of dicts; we then collect both the name and the attribute values from the object (and using the list)

Reply:
Updated for review